### DISCUSSION TITLE
Freewallet/Xchain Music Player with Subassets

### DISCUSSION THREAD

**Discussion by davestaxcp** [jdogresorg/freewallet-desktop#175] [Ideas]
# Freewallet/Xchain Music Player with Subassets

Kane Mayfield was throwing around an idea and was wondering how it might look using xchain and Freewallet.

The idea is to not only being able to play a single song and have it shown when scrolling through your Freewallet, but for a full album to automatically play.

---

Since xchain is resistant to include HTML for various reasons.... Is it possible to link together subassets under a specific token asset name and have them "play through the album of subassets" using xchain (and thus Freewallet)

---

Lets take an example scenario:

- User "MFDOOM" mints the token: **MMFOOD**

then issues each subasset as a song (using easyasset.art or inscribe.art etc. etc.)

so the tracklist would be:

**MMFOOD.BeefRapp**
**MMFOOD.HoeCakes**
**MMFOOD.Potholderz_featuring_Count_Bass_D**

.... and so on

Is it possible to be able to show the asset MMFOOD, or include something in the description of MMFOOD to have it show on xchain (and thus Freewallet) as a full album? 

So if a user loads MMFOOD on their Freewallet User Interface that the whole album can be listened to at once? This would mean when one song (subasset) ends, the next one begins without delay.

---

This could also be used to display playlists, add album songs to the album later, have bonus tracks for the Token version etc.

This could also theorize that the artist could share/send the ALBUM as whole in itself, or the specific individual songs themselves. And some songs could be "rarer" than others.

---

- Would the process be in which how easyasset.art is coding their arweave file? 

- Or with inscribe.art and how you code the inscription?

- How would this work with [CIP25](https://github.com/CounterpartyXCP/cips/blob/master/cip-0025.md)?

---

**BONUS question - Is there a way to do this while also being Token Access Controlled?
^ so the owner of the song or album can access this song within the wallet by signing a message?

---

**Comment by KaneMayfield** [comment]
Yes... and also have it able to play in freewallet.

I understand the liabilities outweigh the benefits when it comes to html... so is there a workaround?

Something that allows me to deploy on xcp... I have a bitcoin infused album I'd like to deploy

**Comment by zerog975** [comment]
Sounds pretty awesome if it can be done
